<b><h1>Exploratory Data Analysis Of Goverment Press Release Data, On Which The Ministry Classifier Is To Be Made For Smart India Hackathon</h1></b>

In [36]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import os
import glob

In [37]:
print(os.listdir('/kaggle/input/press-release'))

['press_release_2003.csv', 'press_release_2005.csv', 'press_release_2004.csv']


In [38]:
df1 = pd.read_csv('../input/press-release/press_release_2003.csv', encoding='unicode_escape')
df2 = pd.read_csv('../input/press-release/press_release_2004.csv', encoding='unicode_escape')
df3 = pd.read_csv('../input/press-release/press_release_2005.csv', encoding='unicode_escape')

In [39]:
df1.head()

pr_id                 pr_datetime                   pr_issued_by  \
0     29  08-December-2003 11:52 IST           Ministry of Railways   
1     35  08-December-2003 12:24 IST  Ministry of Rural Development   
2     39  08-December-2003 12:46 IST              Ministry of Steel   
3     40  08-December-2003 12:47 IST              Ministry of Steel   
4     41  08-December-2003 13:26 IST           Ministry of Railways   

                                            pr_title  \
0  STEAM LOCOMOTIVES FOR PROMOTION OF HERITAGE TO...   
1  ADDITIONAL CENTRAL ASSISTANCE TO ASSAM TO REST...   
2         UPGRADATION OF EXISTING FACILITIES BY SAIL   
3             EXPANSION OF VISAKHAPATNAM STEEL PLANT   
4               EMPHASIS ON USE OF HINDI IN RAILWAYS   

                                          pr_content  
0  Indian Railways have earmarked Nilgiri Mountai...  
1  Ministry of Rural Development has released fun...  
2  LOK SABHA Steel Authority of India Limited (SA...  
3  LOK SABHA There is a proposal for installation...  
4  The Chairman, Railway Board, Shri R.K.Singh ha...

In [40]:
df2.head()

pr_id                pr_datetime                         pr_issued_by  \
0    556  01-January-2004 12:41 IST  Ministry of Housing & Urban Affairs   
1    557  01-January-2004 12:48 IST                 Ministry of Railways   
2    559  01-January-2004 14:31 IST        Ministry of Rural Development   
3    560  01-January-2004 14:29 IST                    Ministry of Steel   
4    561  01-January-2004 14:35 IST      Ministry of Commerce & Industry   

                                            pr_title  \
0  18 new projects worth Rs 122.12 crore for Nort...   
1                        New Secretary Railway Board   
2   Additional allocation of foodgrains to Karnataka   
3                      Steel production shows growth   
4  Exports back on double digit track in November...   

                                          pr_content  
0  Ministry of Urban Development & Poverty Allevi...  
1  Shri V.N.Mathur, presently working as Adviser ...  
2  Ministry of Rural Development has released 1.6...  
3  The total production of steel from April-Novem...  
4  India’s exports during November, 2003 are valu...

In [41]:
df3.head()

pr_id                pr_datetime                           pr_issued_by  \
0   6207  01-January-2005 15:44 IST  Ministry of Health and Family Welfare   
1   6208  01-January-2005 15:46 IST             Ministry of Communications   
2   6209  01-January-2005 16:28 IST                   Ministry of Shipping   
3   6210   01-January-2005 18:1 IST                      Ministry of Mines   
4   6211   01-January-2005 16:4 IST                Ministry of Agriculture   

                                            pr_title  \
0  An additional central medical team leaves for ...   
1  Communication link restored at Campbell Bay an...   
2  T.R. Baalu sanctions 54 new road projects wort...   
3                              Mineral production up   
4          PIB launches Media Centre at Nagapattinam   

                                          pr_content  
0  The Centre is sending an additional team of ei...  
1  The Minister of Communications & Information T...  
2  The Union Minister of Shipping, Road Transport...  
3  The total value of mineral production covering...  
4  Press Information Bureau, Chennai, has set up ...

In [42]:
df1.describe(include='object')

pr_datetime          pr_issued_by          pr_title  \
count                          481                   481               481   
unique                         473                    43               475   
top     11-December-2003 17:36 IST  Ministry of Railways  Press Communique   
freq                             2                    60                 5   

                                               pr_content  
count                                                 481  
unique                                                473  
top     Indian exports of cars and two wheelers have d...  
freq                                                    2

In [43]:
list_of_ministry_1 = df1['pr_issued_by'].unique()
for i in list_of_ministry_1:
    print(i)

Ministry of Railways
Ministry of Rural Development
Ministry of Steel
Ministry of Science & Technology
Ministry of Information & Broadcasting
Ministry of Food Processing Industries
Ministry of Health and Family Welfare
Ministry of Human Resource Development
Ministry of Agriculture
Ministry of Environment and Forests
Ministry of Chemicals and Fertilizers
Ministry of Water Resources
Ministry of Defence
Ministry of Petroleum & Natural Gas
President's Secretariat
Ministry of Micro,Small & Medium Enterprises
Ministry of Mines
Ministry of Tourism
Ministry of Housing & Urban Affairs
Ministry of Coal
Prime Minister's Office
Ministry of Textiles
Ministry of Commerce & Industry
Ministry of External Affairs
Ministry of Social Justice & Empowerment
Ministry of Power
Ministry of Consumer Affairs, Food & Public Distribution
Ministry of Heavy Industries & Public Enterprises
Ministry of Communications
Ministry of Shipping
Ministry of Finance
Ministry of Tribal Affairs
Ministry of Statistics & Programme

<b>There are `43` unique ministry in the data of the 2003</b>

In [44]:
df2.describe(include='object')

pr_datetime          pr_issued_by    pr_title  \
count                         5254                  5254        5061   
unique                        5122                    59        4643   
top     06-December-2004 17:40 IST  Ministry of Railways  Invitation   
freq                             3                   554         169   

                                               pr_content  
count                                                5229  
unique                                               5175  
top     Click here for the Document The Document is in...  
freq                                                    9

In [45]:
list_of_ministry_2 = df2['pr_issued_by'].unique()
for i in list_of_ministry_2:
    print(i)

Ministry of Housing & Urban Affairs
Ministry of Railways
Ministry of Rural Development
Ministry of Steel
Ministry of Commerce & Industry
Ministry of Agriculture
Ministry of Water Resources
Ministry of Tribal Affairs
Prime Minister's Office
Ministry of Finance
Ministry of Mines
Ministry of Communications
Ministry of Tourism
Ministry of Law & Justice
Ministry of Statistics & Programme Implementation
Ministry of Petroleum & Natural Gas
Special Service and Features
Ministry of Social Justice & Empowerment
Ministry of Defence
Ministry of Textiles
Ministry of Consumer Affairs, Food & Public Distribution
President's Secretariat
Ministry of Civil Aviation
Ministry of Information & Broadcasting
Vice President's Secretariat
Ministry of Coal
Ministry of Human Resource Development
Ministry of Home Affairs
Ministry of Agro & Rural Industries
Ministry of Chemicals and Fertilizers
Ministry of New and Renewable Energy
Ministry of Labour & Employment
Ministry of Environment and Forests
UPSC
Ministry of

<b>There are `59` unique ministry in the data of the 2003</b>

In [46]:
df3.describe(include='object')

pr_datetime         pr_issued_by  \
count                      3492                 3492   
unique                     3433                   56   
top     11-March-2005 17:43 IST  Ministry of Finance   
freq                          3                  277   

                                    pr_title  \
count                                   3082   
unique                                  2966   
top     Prime Minister's Engagement Bulletin   
freq                                      26   

                                               pr_content  
count                                                3474  
unique                                               3432  
top     INVITATION Dear Sir/Madam, Â Â Â Â Â Â Â Â Â Â...  
freq                                                    4

In [47]:
list_of_ministry_2 = df2['pr_issued_by'].unique()
for i in list_of_ministry_2:
    print(i)

Ministry of Housing & Urban Affairs
Ministry of Railways
Ministry of Rural Development
Ministry of Steel
Ministry of Commerce & Industry
Ministry of Agriculture
Ministry of Water Resources
Ministry of Tribal Affairs
Prime Minister's Office
Ministry of Finance
Ministry of Mines
Ministry of Communications
Ministry of Tourism
Ministry of Law & Justice
Ministry of Statistics & Programme Implementation
Ministry of Petroleum & Natural Gas
Special Service and Features
Ministry of Social Justice & Empowerment
Ministry of Defence
Ministry of Textiles
Ministry of Consumer Affairs, Food & Public Distribution
President's Secretariat
Ministry of Civil Aviation
Ministry of Information & Broadcasting
Vice President's Secretariat
Ministry of Coal
Ministry of Human Resource Development
Ministry of Home Affairs
Ministry of Agro & Rural Industries
Ministry of Chemicals and Fertilizers
Ministry of New and Renewable Energy
Ministry of Labour & Employment
Ministry of Environment and Forests
UPSC
Ministry of

<b>There are `56` unique ministry in the data of the 2003</b>

<h3>Concatinating all the data frames</h3>

In [48]:
df = pd.concat([df1, df2, df3], ignore_index=True)
df.head()
final_ministry_list = df['pr_issued_by'].unique()

In [49]:
df.describe(include='object')

pr_datetime          pr_issued_by    pr_title  \
count                         9227                  9227        8624   
unique                        9028                    60        8047   
top     06-December-2004 17:40 IST  Ministry of Railways  Invitation   
freq                             3                   812         169   

                                               pr_content  
count                                                9184  
unique                                               9076  
top     Click here for the Document The Document is in...  
freq                                                    9

<b>So total number of rows/ press release that we are having is: `9227`, We need more press releases from the goverment and for that the web crawling is not allowed in the website, So we have to options for now:
<ul>
       <li>Do mail to authority data person of PIB to provide us data of all press releases</li>
       <li>Do an RTI of it, So we will get the data</li>
       <li>Make an person to seat and take the data from website and put in excel sheet</li>
</ul>
</b>

In [50]:
df.isnull().sum()

pr_id             0
pr_datetime       0
pr_issued_by      0
pr_title        603
pr_content       43
dtype: int64

As out motive is to get the ministry related content from the given data and it's related new, So our work will be not affected by the null values of the title and the content, So we will make no change in them

<h3>Now we will extract the numbers of the articles belonging to each ministry</h3>

In [51]:
ministry_number_pr_dict = {}

for i in final_ministry_list:
    print(i)
    print(df[df['pr_issued_by']==i]['pr_id'].count())
    ministry_number_pr_dict[i] = df[df['pr_issued_by']==i]['pr_id'].count()
    print('-\n')

Ministry of Railways
812
-

Ministry of Rural Development
165
-

Ministry of Steel
146
-

Ministry of Science & Technology
181
-

Ministry of Information & Broadcasting
240
-

Ministry of Food Processing Industries
50
-

Ministry of Health and Family Welfare
215
-

Ministry of Human Resource Development
289
-

Ministry of Agriculture
263
-

Ministry of Environment and Forests
155
-

Ministry of Chemicals and Fertilizers
113
-

Ministry of Water Resources
146
-

Ministry of Defence
421
-

Ministry of Petroleum & Natural Gas
168
-

President's Secretariat
226
-

Ministry of Micro,Small & Medium Enterprises
83
-

Ministry of Mines
237
-

Ministry of Tourism
91
-

Ministry of Housing & Urban Affairs
88
-

Ministry of Coal
131
-

Prime Minister's Office
760
-

Ministry of Textiles
76
-

Ministry of Commerce & Industry
511
-

Ministry of External Affairs
211
-

Ministry of Social Justice & Empowerment
158
-

Ministry of Power
81
-

Ministry of Consumer Affairs, Food & Public Distribution
116

In [52]:
ministry_number_pr_dict.items()

dict_items([('Ministry of Railways', 812), ('Ministry of Rural Development', 165), ('Ministry of Steel', 146), ('Ministry of Science & Technology', 181), ('Ministry of Information & Broadcasting', 240), ('Ministry of Food Processing Industries', 50), ('Ministry of Health and Family Welfare', 215), ('Ministry of Human Resource Development', 289), ('Ministry of Agriculture', 263), ('Ministry of Environment and Forests', 155), ('Ministry of Chemicals and Fertilizers', 113), ('Ministry of Water Resources', 146), ('Ministry of Defence', 421), ('Ministry of Petroleum & Natural Gas', 168), ("President's Secretariat", 226), ('Ministry of Micro,Small & Medium Enterprises', 83), ('Ministry of Mines', 237), ('Ministry of Tourism', 91), ('Ministry of Housing & Urban Affairs', 88), ('Ministry of Coal', 131), ("Prime Minister's Office", 760), ('Ministry of Textiles', 76), ('Ministry of Commerce & Industry', 511), ('Ministry of External Affairs', 211), ('Ministry of Social Justice & Empowerment', 158

<h2>We have to also add the sentiment of the news in this dataset</h2>

In [53]:
df.head(10)

pr_id                 pr_datetime                            pr_issued_by  \
0     29  08-December-2003 11:52 IST                    Ministry of Railways   
1     35  08-December-2003 12:24 IST           Ministry of Rural Development   
2     39  08-December-2003 12:46 IST                       Ministry of Steel   
3     40  08-December-2003 12:47 IST                       Ministry of Steel   
4     41  08-December-2003 13:26 IST                    Ministry of Railways   
5     42  08-December-2003 13:45 IST        Ministry of Science & Technology   
6     43  08-December-2003 13:52 IST        Ministry of Science & Technology   
7     44  08-December-2003 15:10 IST  Ministry of Information & Broadcasting   
8     45  08-December-2003 15:16 IST                       Ministry of Steel   
9     46  08-December-2003 15:49 IST  Ministry of Food Processing Industries   

                                            pr_title  \
0  STEAM LOCOMOTIVES FOR PROMOTION OF HERITAGE TO...   
1  ADDITIONAL CENTRAL ASSISTANCE TO ASSAM TO REST...   
2         UPGRADATION OF EXISTING FACILITIES BY SAIL   
3             EXPANSION OF VISAKHAPATNAM STEEL PLANT   
4               EMPHASIS ON USE OF HINDI IN RAILWAYS   
5  INDIA CALLS FOR CLOSER DIALOGUE AND INTERACTIO...   
6  INDIA AGAINST PATENT LAWS NOT BENEFICIAL TO HU...   
7  GOVERNMENT COMMITTED FOR OVERALL DEVELOPMENT O...   
8                           PROPOSAL TO REVIVE IISCO   
9  FINANCIAL ASSISTANCE TO STATES FOR SETTING UP ...   

                                          pr_content  
0  Indian Railways have earmarked Nilgiri Mountai...  
1  Ministry of Rural Development has released fun...  
2  LOK SABHA Steel Authority of India Limited (SA...  
3  LOK SABHA There is a proposal for installation...  
4  The Chairman, Railway Board, Shri R.K.Singh ha...  
5  India has called for closer dialogue and inter...  
6  India is against patent laws that are not bene...  
7  Rajya Sabha The Government has taken several i...  
8  LOK SABHA Government has approved a proposal f...  
9  LOK SABHA The Ministry of Food Processing Indu...

In [54]:
df.iloc[0]['pr_content']

'Indian Railways have earmarked Nilgiri Mountain Railways on metre guage and Darjeeling Himalayan Railways on narrow guage for promotion of heritage tourism by steam locomotive. During the year-long celebrations of 150 glorious years of railways in India, steam loco hauled trains were run on Mumbai-Thane, Shimla-Kathleeghat, Bolarm-Monoharabad and Howrah-Bandel sections by reviving steam locomotives. At present, 33 steam locomotives are working on Indian Railways.'

<b>Analyzing the problem of null, What types of the fields are null, And removing the rows that have both title and content null</b>

In [55]:
null_titles = df[df['pr_title'].isnull()]
print(len(null_titles))

null_contents = df[df['pr_content'].isnull()]
print(len(null_contents))

null_boths = df[df['pr_title'].isnull() & df['pr_content'].isnull()]
print(len(null_boths))

603
43
0


<b>In the dataset, There is no condition such that, Both the title and content are null, So that there is no concern regarding it, If it is there one have to also write the code to remove the rows in which both are empty or null.</b>

In [56]:
null_contents.head(43)

pr_id                 pr_datetime             pr_issued_by  \
815     916  03-February-2004 16:15 IST      Ministry of Finance   
816     917  03-February-2004 16:16 IST      Ministry of Finance   
817     918  03-February-2004 16:32 IST      Ministry of Finance   
818     919  03-February-2004 16:32 IST      Ministry of Finance   
819     920  03-February-2004 16:33 IST      Ministry of Finance   
820     921  03-February-2004 16:38 IST      Ministry of Finance   
821     922  03-February-2004 16:40 IST      Ministry of Finance   
2120   2313      07-July-2004 14:16 IST      Ministry of Finance   
2121   2314      07-July-2004 14:19 IST      Ministry of Finance   
2122   2315      07-July-2004 14:32 IST      Ministry of Finance   
2123   2316      07-July-2004 14:32 IST      Ministry of Finance   
2124   2317      07-July-2004 14:23 IST      Ministry of Finance   
2125   2318      07-July-2004 14:24 IST      Ministry of Finance   
2126   2319      07-July-2004 14:25 IST      Ministry of Finance   
2127   2320      07-July-2004 14:26 IST      Ministry of Finance   
2128   2321      07-July-2004 14:27 IST      Ministry of Finance   
2129   2322      07-July-2004 14:28 IST      Ministry of Finance   
2130   2323      07-July-2004 14:30 IST      Ministry of Finance   
2184   2387      08-July-2004 17:41 IST      Ministry of Finance   
2185   2388      08-July-2004 15:18 IST      Ministry of Finance   
2186   2389      08-July-2004 14:59 IST      Ministry of Finance   
2187   2390       08-July-2004 15:0 IST      Ministry of Finance   
2188   2391       08-July-2004 15:2 IST      Ministry of Finance   
2189   2392       08-July-2004 15:3 IST      Ministry of Finance   
2190   2393       08-July-2004 15:4 IST      Ministry of Finance   
6223   6724   24-January-2005 20:38 IST  Ministry of Agriculture   
6763   7326  25-February-2005 14:25 IST      Ministry of Finance   
6767   7330  25-February-2005 14:48 IST      Ministry of Finance   
6768   7331  25-February-2005 14:51 IST      Ministry of Finance   
6769   7332  25-February-2005 14:55 IST      Ministry of Finance   
6770   7333  25-February-2005 14:56 IST      Ministry of Finance   
6771   7334  25-February-2005 15:14 IST      Ministry of Finance   
6772   7335   25-February-2005 15:4 IST      Ministry of Finance   
6773   7336   25-February-2005 15:6 IST      Ministry of Finance   
6774   7337  25-February-2005 15:11 IST      Ministry of Finance   
6860   7424  28-February-2005 17:40 IST      Ministry of Finance   
6861   7425  28-February-2005 17:43 IST      Ministry of Finance   
6862   7426  28-February-2005 17:46 IST      Ministry of Finance   
6863   7427  28-February-2005 17:47 IST      Ministry of Finance   
6864   7428  28-February-2005 17:48 IST      Ministry of Finance   
6865   7429  28-February-2005 17:50 IST      Ministry of Finance   
6866   7430  28-February-2005 17:51 IST      Ministry of Finance   
6867   7431  28-February-2005 17:52 IST      Ministry of Finance   

                                               pr_title pr_content  
815                                 Deficit as % of GDP        NaN  
816                                    Rupee comes from        NaN  
817                                       Rupee goes to        NaN  
818                                            Receipts        NaN  
819                      Central Plan Outlay by Sectors        NaN  
820                     Resources transferred to States        NaN  
821                                   Total Expenditure        NaN  
2120  Export of Principle Commodities ( % Share ), 2...        NaN  
2121                          Foreign Direct Investment        NaN  
2122                          Foreign Exchange Reserves        NaN  
2123                                   GDP Growth Rate*        NaN  
2124                                Imports and Exports        NaN  
2125  Import of Principal Commodities (% Share),2003-04        NaN  
2126                      Performance Of Teleco

<b>Including the rows with the only title, As the titles makes the logical sense with the minsitry, And will positively imapact the data</b>

In [57]:
null_titles.head()

pr_id                pr_datetime  \
4173   4561  28-October-2004 15:36 IST   
4176   4564  28-October-2004 16:25 IST   
4178   4566  28-October-2004 16:46 IST   
4182   4570  28-October-2004 18:19 IST   
4191   4579  28-October-2004 11:14 IST   

                                pr_issued_by pr_title  \
4173  Ministry of Information & Broadcasting      NaN   
4176                       Ministry of Power      NaN   
4178     Ministry of Housing & Urban Affairs      NaN   
4182  Ministry of Human Resource Development      NaN   
4191              Ministry of Civil Aviation      NaN   

                                             pr_content  
4173  INVITATION Dear Sir/Madam, You are cordially i...  
4176  I N V I T A T I O N Dear Sir/Madam, FUNCTION: ...  
4178  INVITATION VISUAL COVERAGE ONLY Sir/Madam, Kin...  
4182  INVITATION Dear Sir/Madam, The Human Resource ...  
4191  INVITATION Dear Sir/Madam, Kindly arrange cove...

In [58]:
null_titles.iloc[0]['pr_content']

'INVITATION Dear Sir/Madam, You are cordially invited to cover the following function: FUNCTION : The Minister of Information & Broadcasting and Culture, Shri S. Jaipal Reddy will inaugurate the Second Conference of Editors & Working Journalists of SAARC Countries. VENUE : Hall No. 6, Vigyan Bhawan, New Delhi. DATE : 28th October, 2004 (Thursday). TIME : 10.00 a.m. Yours sincerely, (N.N. KAUL) To All Accredited correspondents All Accredited electronic media/still camerapersons ADG (News) AIR ADG (News) Doordarshan DPR (Photo) 3 copies \x96 kindly release the photo'

In [59]:
null_titles.iloc[1]['pr_content']

'I N V I T A T I O N Dear Sir/Madam, FUNCTION: The Union Minister of Power, Shri P.M. Sayeed will address the CII Annual National Conference 2004, Plenary 4: Infrastructure. VENUE: Hotel The Grand, Vasant Kunj, New Delhi. DATE: 28.10.2004 TIME: 1430 hrs. You are requested to cover the event. ( ESTHER KAR ) All accredited Correspondents/Camera Persons (still & movie). ADG (News), Doordarshan/AIR Reporting. All TV Channels. DPR (Photo) \x96 2 copies/Director (Photo Division) \x96 1 copy.'

In [60]:
null_titles.iloc[500]['pr_content']

'INVITATION Dear Sir/ Madam,            Dr. A.P.J. Abdul Kalam, President of India will be the Chief Guest at the inaugural session of 11th National Water Convetion. Shri Priyaranjan Dasmunsi, Minister of Water Resources will preside over the function while Shri Jai Prakash Narayan Yadav, Minister of State for Water Resources will be Guest of Honour as per the following programme:                                     FUNCTION : 11th National Water Convention VENUE : Vigyan Bhavan, New Delhi DATE : Wednesday, 11th May, 2005 TIME : 12:15 Noon            You are cordially invited to cover the function.           Yours sincerely,  (RAJEEV JAIN) To All Correspondents All electronic media/still camerapersons ADG (News) AIR ADG (News) Doordarshan DPR (Photo) 3 copies â<U+0080><U+0093> kindly release the photo NB â<U+0080><U+0093> 1. You are requested to be seated by 11.45 AM. 2. Entry by PIB cards. Invitation cards and pa

<b>Currenly including the conents without the title, As they have very less logic of invitation, But message contains the name of the mistry, Will compare the accuracy, By making one model with it and other without it....</b>

In [61]:
df.iloc[815]

pr_id                                  916
pr_datetime     03-February-2004 16:15 IST
pr_issued_by           Ministry of Finance
pr_title               Deficit as % of GDP
pr_content                             NaN
Name: 815, dtype: object

<b>Removing the `9227-9184=43`, Rows which does not contains the title and content,As they are only improtant to us, If they are not there then they are of not of any use</b>

In [62]:
max = 0
idx = -1
for i in range(len(df)):
    try:
        if (max < len(df.loc[i, 'pr_content'])):
            max = len(df.loc[i, 'pr_content'])
            idx = i
    except:
        pass
print(max)
print(idx)
print(df.loc[idx, 'pr_title'],'------')
print(df.loc[idx, 'pr_issued_by'],'------')
print(df.loc[idx, 'pr_content'])

32759
717
Republic Day Police Medals, 2004 announced POLICE MEDALS TO 627 PERSONNEL ------
Ministry of Home Affairs ------
The President is pleased to award this year’s Republic Day Police Medals to 627 Police Personnel including 76 Distinguished Service Medals & 104 Gallantry Medals. The list of awardees is as follows:- PRESIDENT’S POLICE MEDAL FOR DISTINGUISHED SERVICE REPUBLIC DAY – 2004 PRESIDENT’S POLICE MEDAL FOR DISTINGUISHED SERVICE – REPUBLIC DAY- 2004 SHRI A K SHRIVASTAVA, ADDL.DGP, HYDERABAD, ANDHRA PRADESH SHRI ALOK SRIVASTAVA, DIRECTOR, ACB, HYDERABAD, ANDHRA PRADESH SHRI M ALAGAR, ADDL.CP. ( CRIME & SIT), HYDERABAD, ANDHRA PRADESH SHRI A SHANKAR, COMMANDANT, 7TH BN APSP, DICHPALLY, ANDHRA PRADESH SHRI N SRINIVAS, ADDL.DCP, CSW, HYDERABAD, ANDHRA PRADESH SHRI J S YADAV, INSPECTOR / OC PS, ITANAGAR, ARUNACHAL PRADESH SHRI RAJENDRA SINGH, INSPECTOR, EKANGAR SARAI(NALANDA), BIHAR SHRI SUBHASH CHANDRA ATRE, IGP SBI(E.O), RAIPUR, CHHATTISGARH SMT PRATIMA SHARMA, INSPECTOR/ SHO,

In [63]:
df.loc[103, 'pr_content']

'Shri S.K. Dubey, Dy. General Manager (Technical), National Highways Authority of India (NHAI), posted at Gaya, was murdered in the early hours of 27 November, 2003. A series of articles and news items have appeared on this unfortunate incident. These have understandably created a sense of shock and an outpouring of sympathy for Shri Dubey. His loss is deeply mourned by his colleagues in NHAI as well as by the Ministry of Road Transport & Highways. In his tribute to Shri Dubey, honourable Prime Minister has called him "an upright and dedicated officer" and assured the nation that "those responsible for his death, wherever they may be, will not be spared". The case has already been handed over to the CBI for an impartial enquiry. However, a section of the media, without highlighting all the relevant facts of the matter, has drawn misleading inferences on the cause of Shri Dubey\x92s murder. Its coverage has also sought to project the functioning of the NHAI and the Government in a negat

In [64]:
df.loc[54, 'pr_content']

'Forty nine Master Craftspersons and Weavers have been selected from all over the country for National Awards for the years 2000 and 2001. 29 of these National Award winners have been selected from the Handicrafts sector and 20 from the Handlooms sector. Seven out of the 49 winners are joint awardees. The award carries Rs.25,000/- a Tamrapatra, Angavastram and a certificate. Twenty eight craftspersons and seven weavers have also been selected for National Merit Certificates. The winners of the Merit Certificates will receive Rs.5000/- in cash and a certificate. The President Dr. A.P.J. Abdul Kalam will confer the awards at a function in New Delhi on the 12th of this month. The Award and Certificate winners belong to such diverse crafts as Papier Machie, Kalamkari Painting, Miniature Painting, Banjara Embroidery, Thewa Art, Mithila Painting, Gold Leaf Painting, Jewellery, Longpi Pottery, Cane and Bamboo, Pattachitra and Sea Shell Craft etc. The Master weavers selected for the awards and

<h2>As demonstrated that the content legth is much larger and it does not contains the information of any intrest after some characters, So we will only include some starting characters of content only...</h2>

In [65]:
max = 0
idx = -1
for i in range(len(df)):
    
    if i==717 or i==103:
        break
    
    try:
        if (max < len(df.loc[i, 'pr_title'])):
            max = len(df.loc[i, 'pr_title'])
            idx = i
    except:
        pass
print(max)
print(idx)
print(df.loc[idx, 'pr_title'],'------')
print(df.loc[idx, 'pr_issued_by'],'------')
print(df.loc[idx, 'pr_content'])

# 717, 103

281
74
India calls for removal of barriers for transfer of climate friendly technology at COP-9 Meeting at Milan "All facets of sustainable development must contribute positively to povery alleviation"-T.R.Baalu India and the United States identify cutting edge technology for cooperation ------
Ministry of Environment and Forests ------
India has called for removal of obstacles- financial and IPRs, for the successful adaptation of emerging technology options in energy efficiency, fuel switching, and renewable energy. Taking part at the First Round Table of the High Level segment of COP-9 at Milan last night, the Environment and Forests Minister, Shri T.R.Baalu said that serious action to overcome these barriers is needed, without further delay, as the adverse impact of climate change is already being felt by poor and developing countries who are most vulnerable to extreme climate events such as floods, droughts and severe heat waves. Talking on the issue of sustainable development, Shr

In [68]:
str = "India calls for removal of barriers for transfer of climate friendly technology at COP-9 Meeting at Milan All facets of sustainable development must contribute positively to povery alleviation-T.R.Baalu India and the United States identify cutting edge technology for cooperation India has called for removal of obstacles- financial and IPRs, for the successful adaptation of emerging technology options in energy efficiency, fuel switching, and renewable energy. Taking part at the First Round Table of the High Level segment of COP-9 at Milan last night, the Environment and Forests Minister"

In [69]:
len(str)

593

<b>Maximum title is of the `281` length of characters, so there is no problem in it, It is normal</b>

<h2>So for out prediction, We will only consider the 550 first characters including the length of title and content for prediction in our model</h2>

In [73]:
str1 = "India has called for removal of obstacles- financial and IPRs, for the successful adaptation of emerging technology options in energy efficiency, fuel switching, and renewable energy. Taking part at the First Round Table of the High Level segment of COP-9 at Milan last night, the Environment and Forests Minister, Shri T.R.Baalu said that serious action to overcome these barriers is needed, without further delay, as the adverse impact of climate change is already being felt by poor and developing countries who are most vulnerable to extreme climate events such as floods, droughts and severe heat waves. Talking on the issue of sustainable development, Shri Baalu said that it involves realization of synergies and complementarities between economic growth and environmental protection. These in turn are dependent on several factors such as technological change, trade regimes, historical patterns of access to global resources and national policies. As far as India is concerned, all facets of sustainable development must contribute positively to the dominant challenge of our times-that of poverty alleviation, Shri Baalu said. Citing various projections about Green House Gas emissions from the developed countries, Shri Baalu said that it may increase by 10 per cent over the 1990 levels by 2010. Emissions from countries with economies in transition are also starting to increase. Effective action needs to be taken by the industrialized countries for curtailing GHG emissions both domestically and through cooperative mechanisms, he said. Turning to the issue of adaptation, Shri Baalu said, it must enhance the ability of developing countries to effectively cope with unavoidable climate changes. Adaptation measures must be embodied in their overall sustainable strategies. Reiterating the need for early operationalisation of the special climate fund and the LDC fund as agreed at COP-8, Shri Baalu said that adaptation measures need to be supported and facilitated through financial and technology transfers by Annexure-II parties under their United Nations Framework Convention On Climate Change (UNFCCC) commitments. Opening the ministerial round table discussions, the President of COP-9 and the Minister of Environmental Protection and Water Management of Hungary, Mr.Miklos Persanyi paid glowing tributes to the initiative of Shri Baalu for bringing the linkage between climate change and sustainable development to the fore at the COP-8 meeting in New Delhi last year. He said the twin issues of technology transfer and adaptation, the major concerns of the developing countries have been pursued by Shri Baalu during the inter-sessional period and concretized in the recent Climate Technology Bazaar and Conferences on adaptation held in New Delhi last month. On the margins of COP-9 Shri Baalu also had bilateral discussions with US Under Secretary of State for Global Affairs and the leader of the US delegation, Ms. Paula Dobriansky on Indo-US cooperation on climate change. Several areas like climate modeling, early warning systems, energy efficiency and renewables were identified for bilateral cooperation. Shri Baalu said that the future projects should focus on the needs of the vulnerable sections of society and adaptation concerns must be integrated to them. He also suggested that a specific project may be taken up on vulnerability assessment and adaptation in stretches in Tamil Nadu in cooperation with local communities like fishermen to study climate change propelled rise in sea level and weather related events such as cyclones. Ms. Dobriansky gave an account of the Research and Development being carried out in the United States towards a Hydrogen Economy. While appreciating the objectives of the Project Shri Baalu said that any research and development should lead to the technology being socio-economically viable and within the reach of common man. He said that a cost benefit analysis of the programme should be taken up and added that vulnerable sections should be able to make use of the technology for operating small pump sets and satisfying their basic needs like drawing water."
print(str1[:550])

India has called for removal of obstacles- financial and IPRs, for the successful adaptation of emerging technology options in energy efficiency, fuel switching, and renewable energy. Taking part at the First Round Table of the High Level segment of COP-9 at Milan last night, the Environment and Forests Minister, Shri T.R.Baalu said that serious action to overcome these barriers is needed, without further delay, as the adverse impact of climate change is already being felt by poor and developing countries who are most vulnerable to extreme clim
